controllers/cluster_controller_test.go

```diff
@@ -64,6 +64,43 @@ func reloadCluster(cluster *fdbv1beta2.FoundationDBCluster) (int64, error) {
 	return generations.Reconciled, err
 }
 
+func generationsNotMatching(cluster *fdbv1beta2.FoundationDBCluster) func() {
+	return func() {
+		generations := reloadClusterGenerationsSafe(cluster)
+		Expect(generations.Reconciled).ToNot(Equal(cluster.ObjectMeta.Generation))
+	}
+}
+
+func generationsMatching(cluster *fdbv1beta2.FoundationDBCluster) func() {
+	return func() {
+		generations := reloadClusterGenerationsSafe(cluster)
+		Expect(generations.Reconciled).To(Equal(cluster.ObjectMeta.Generation))
+	}
+}
+
+func reloadClusterGenerationsSafe(cluster *fdbv1beta2.FoundationDBCluster) fdbv1beta2.ClusterGenerationStatus {
+	status, err := reloadClusterGenerations(cluster)
+	Expect(err).NotTo(HaveOccurred())
+	return status
+}
+
+func updateStorageEngineVersion(cluster *fdbv1beta2.FoundationDBCluster, storageEngine fdbv1beta2.StorageEngine, version string) func() {
+	return func() {
+		cluster.Spec.DatabaseConfiguration.StorageEngine = storageEngine
+		cluster.Spec.Version = version
+		Expect(k8sClient.Update(context.TODO(), cluster)).NotTo(HaveOccurred())
+	}
+}
+
+func updateStorageEngineVersionIncompleteReconcilliation(cluster *fdbv1beta2.FoundationDBCluster, storageEngine fdbv1beta2.StorageEngine, version string, shouldCompleteReconciliation *bool) func() {
```

Comment:
We can adjust this method to not return a closure but actually perform all the steps.